gantry/routes/prediction/prediction.py

```diff
@@ -0,0 +1,192 @@
+import logging
+import os
+
+import aiosqlite
+
+from gantry.routes.prediction.current_mapping import pkg_mappings
+from gantry.util import k8s
+
+logger = logging.getLogger(__name__)
+
+IDEAL_SAMPLE = 5
+DEFAULT_CPU_REQUEST = 1.0
+DEFAULT_MEM_REQUEST = 2 * 1_000_000_000  # 2GB in bytes
+EXPENSIVE_VARIANTS = {
+    "sycl",
+    "mpi",
+    "rocm",
+    "cuda",
+    "python",
+    "fortran",
+    "openmp",
+    "hdf5",
+}
+
+
+async def predict_single(db: aiosqlite.Connection, spec: dict) -> dict:
```

Comment:
```suggestion
async def predict(db: aiosqlite.Connection, spec: dict) -> dict:
```

Since we dropped the multi-spec prediction I'm tempted to say we should rename this to simply be predict.